index.js

```diff
@@ -3044,81 +3044,151 @@
     impl: at
   };
 
-  //# head :: Array a -> Maybe a
+  //# head :: Foldable f => f a -> Maybe a
   //.
-  //. Returns Just the first element of the given array if the array contains
-  //. at least one element; Nothing otherwise.
+  //. Returns Just the first element of the given structure if it
+  //. contains at least one element; Nothing otherwise.
   //.
   //. ```javascript
   //. > S.head ([1, 2, 3])
   //. Just (1)
   //.
   //. > S.head ([])
   //. Nothing
+  //.
+  //. > S.head (Cons ('foo') (Cons ('bar') (Nil)))
+  //. S.Just ('foo')
+  //.
+  //. > S.head (Nil)
+  //. S.Nothing
```

Comment:
For the expected output we like to use `Nothing` and `Just` rather than `S.Nothing` and `S.Just`. This makes the doctests for [`S.match`][1] and several other functions less noisy.


[1]: https://sanctuary.js.org/#match

Reply:
🆗